command.go

```diff
@@ -189,9 +189,11 @@ func (c *Command) parseFlags(args Args, shellComplete bool) (*flag.FlagSet, erro
 		return set, set.Parse(append([]string{"--"}, args...))
 	}
 
+	fmt.Println("before reordering:", strings.Join(args, ","))
 	if !c.SkipArgReorder {
 		args = reorderArgs(c.Flags, args)
 	}
+	fmt.Println("after reordering: ", strings.Join(args, ","))
```

Comment:
Problem is in this area; 

```
=== RUN   TestParseAndRunHyphenValues/foo_test_--opt_-opt-value_arg1
before reordering: --opt,-opt-value,arg1
after reordering:  --opt,arg1,-opt-value
    TestParseAndRunHyphenValues/foo_test_--opt_-opt-value_arg1: helpers_test.go:20: (command_test.go:175) Expected <nil> (type <nil>) - Got flag provided but not defined: -opt-value (type *errors.errorString)
    TestParseAndRunHyphenValues/foo_test_--opt_-opt-value_arg1: helpers_test.go:20: (command_test.go:176) Expected [arg1] (type []string) - Got [] (type []string)
    TestParseAndRunHyphenValues/foo_test_--opt_-opt-value_arg1: helpers_test.go:20: (command_test.go:177) Expected -opt-value (type string) - Got  (type string)
```

looks like the `reorderArgs` does not take into account if an `arg` is a value for a flag, and now `arg1` becomes an argument for `--opt`;

```
after reordering:  --opt,arg1,-opt-value
```